activestorage/app/controllers/active_storage/base_controller.rb

```diff
@@ -9,6 +9,17 @@ class ActiveStorage::BaseController < ActionController::Base
   self.etag_with_template_digest = false
 
   private
+    def stream_chunk(blob)
+      range = Rack::Utils.get_byte_ranges(request.headers["HTTP_RANGE"], blob.byte_size).first
```

Comment:
have you tested what happens if the client requests more than one range? I am not sure how common this is, but it is allowed by the standards, and I am not sure how clients will behave if you return only the first range they requested.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests#multipart_ranges